server/notifications.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package server
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gofrs/uuid"
+)
+
+const (
+	// how many notifications can be in a notification buffer.
+	notificationBufferSize = 16
+)
+
+type change struct {
+	// persistedIndex is a changed persisted index or `nil` if the value has not changed.
+	persistedIndex *uint64
+}
+
+// Any returns true if there is a change.
+func (c change) Any() bool {
+	return c.persistedIndex != nil
+}
+
+type notifications struct {
```

Comment:
This could probably be unit tested on its own to isolate it from the rest of the server implementation.

Reply:
I see the implementation, rather than sending a pointer over a channel couldn't you just simplify the sending logic to something like:

```go
var c *change

if persistedIndex != oldPersistedIndex {
    c = change{persistedIndex}
}

if c != nil {
    go func() {
        serv.notifications.notify(ctx, *c)
    }()
}
```

Reply:
Sorry, I don't understand your [last comment](https://github.com/elastic/elastic-agent-shipper/pull/76#discussion_r921542698). This is how it's implemented at the moment, am I missing something? There is no pointer to the `change` struct either.

https://github.com/rdner/elastic-agent-shipper/blob/d4239c1f02abeeb46402c26226477c6b215c2372/server/server.go#L264-L288

Reply:
Sorry, let's ignore the second comment here. It should have been made on this thread https://github.com/elastic/elastic-agent-shipper/pull/76#discussion_r921540711

Reply:
@cmacknz I don't think this needs a separate unit test, this is not an exported part of the package, it's only used by the server implementation and it's fully covered by the current tests.